File: 1319. Unique Number of Occurrences/Solution.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        freq = {}
        for num in arr:
            freq[num] = freq.get(num, 0) + 1
        
        count_set = set()
        for count in freq.values():
            if count in count_set:
                return False
            count_set.add(count)
        
        return True

File: 1319. Unique Number of Occurrences/Solution.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        std::unordered_map<int, int> freq;
        for (int num : arr) {
            freq[num]++;
        }

        std::unordered_set<int> countSet;
        for (const auto& kv : freq) {
            if (countSet.find(kv.second) != countSet.end()) {
                return false;
            }
            countSet.insert(kv.second);
        }        
        return true;
    }
};